### DISCUSSION TITLE
Applying volumetric constraint in phase field model for whole domain

### DISCUSSION THREAD

**Discussion by vermaprk** [idaholab/moose#26770] [Q&A Modules: Phase field]
Hello
I am trying to model the phase-field variable eta which evolves due to electroplating at a constant current density. I have built the kernels for eta. However, I want an external constraint as follow:
volume integral(d eta/ dt)dV = Constant (which signifies the volume integral of change is order parameter w.r.t time is constant as total current is held constant )
How do I implement such constraint in moose?
Thanks

**Comment by GiudGiud** [comment]
Hello

Have you looked the Constraints system?
Or can you formulate the constraint as a Bound on a variable (with the Bounds system)? THat's also an option

Guillaume

**Reply by vermaprk** [reply]
yes I looked into constraint systems. However, I found some discussion on this at 
https://groups.google.com/g/moose-users/c/7u7z51mZ3xE
The constraint I am trying to apply is not localized for an element but is the volumetric integral over whole simulation box. As per the discussion in the above link I came across ScalarLagrangeMultiplier
But again I am confused whether this constraint is per element or for the whole simulation domain.
Thanks 
My PDE looks like this
![eta](https://github.com/idaholab/moose/assets/69255835/319a4f35-8038-465d-a2d7-190cb5838485)
and constraint
![constraint](https://github.com/idaholab/moose/assets/69255835/f7daa7c1-1ac5-44bf-8d6c-0e8acc290361)
here Ir is constant

**Reply by GiudGiud** [reply]
Are c_s, F, S and xi variables? And Ir is a known constant?
I think we can apply this constraint with a lagrange multiplier otherwise, using a scalar kernel. 
It's been done with this scalar kernel for the average
https://mooseframework.inl.gov/source/scalarkernels/AverageValueConstraint.html

it could be adapted for any integral
